test/model.test.js

```diff
@@ -0,0 +1,893 @@
+    expect(doc1.str).to.equal(str1);
+    expect(doc1).to.not.eql(doc2);
+  });
+
+  it('two instances should have different prototypes', function() {
+    var doc1 = new test.models.Model({ str: util.s8() });
+    var doc2 = new test.models.Model({ str: util.s8() });
+
+    expect(Object.getPrototypeOf(doc1)).to.not.eql(Object.getPrototypeOf(doc2));
+  });
+
+  it('two instances should have the same model', function() {
+    var doc1 = new test.models.Model({ str: util.s8() });
+    var doc2 = new test.models.Model({ str: util.s8() });
+
+    expect(doc1.getModel()).to.eql(doc2.getModel());
+  });
+
+  it('docs from different models should not interfere', function() {
+    var str = util.s8();
+    var doc = new test.models.Model({str: str});
+
+    var otherName = util.s8();
+    test.models.OtherModel = thinky.createModel(otherName, {
+      str: String
+    });
+
+    return test.models.OtherModel.ready()
+      .then(function() {
+        var otherStr = util.s8();
+        var otherDoc = new test.models.OtherModel({ str: otherStr });
+
+        expect(doc.str).to.eql(str);
+        expect(otherDoc.str).to.eql(otherStr);
+
+        expect(otherDoc.getModel()).to.not.eql(doc.getModel());
+
+        // TODO: better way to access schema data, maybe Model.getSchema()
+        expect(doc.getModel().getTableName()).to.eql(test.models.Model._schema._model._name);
+        expect(otherDoc.getModel().getTableName(), otherName);
+      });
+  });
+
+});
+
+describe("Batch insert", function() {
+  afterEach(function() { return test.dropTables(); });
+
+  it('should insert a single document', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number
+    });
+
+    return Model.ready()
+      .then(function() {
+        return Model.save({ id: 'foo' });
+      })
+      .then(function(result) {
+        // NOTE: deepCopy to remove prototype. Otherwise these are
+        //       not, in fact, equivalent objects, even though
+        //       assert will claim they are
+        expect(util.deepCopy(result)).to.eql({ id: 'foo' });
+      });
+  });
+
+  it('should insert a batch of documents', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number
+    });
+
+    var docs = [];
+    for (var i = 0; i < 10; i++) {
+      docs.push({ num: i });
+    }
+
+    return Model.ready()
+      .then(function() {
+        return Model.save(docs);
+      })
+      .then(function(result) {
+        expect(result).to.eql(docs);
+
+        for (i = 0; i < 10; i++) {
+          expect(docs[i].id).to.be.a('string');
+          expect(docs[i].isSaved()).to.be.true;
+        }
+      });
+  });
+
+  it('should validate fields before saving', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number
+    });
+
+    return Model.ready()
+      .then(function() {
+        return Promise.all([
+          expect(Model.save([{id: 4}]))
```

Comment:
Though I'm not sure what `expect(...).to.be.eventually.rejectedWith` is supposed to return